digitalearthau/index.py

```diff
@@ -65,21 +70,23 @@ def add_dataset(index: Index, dataset_id: uuid.UUID, uri: str):
     A better api should be pushed upstream to core: it currently only has a "scripts" implementation
     intended for cli use.
     """
-    path = uri_to_local_path(uri)
-    for d in dataset_script.load_datasets([path], _get_rules(index)):
-        if d.id == dataset_id:
-            index.datasets.add(d, sources_policy='ensure')
-            break
-    else:
-        raise RuntimeError('Dataset not found at path: %s, %s' % (dataset_id, uri))
+    yaml_path = uri_to_local_path(uri)
+
+    ds_resolve = Doc2Dataset(index)
+    doc_stream = ui_path_doc_stream([yaml_path], logger=_LOG, uri=True)
+    dss = dataset_script.dataset_stream(doc_stream, ds_resolve)
+
+    try:
+        dataset_script.index_datasets(dss,
```

Comment:
`index_datasets` isn't doing much, it's really just `map(index.datasets.add, dss)` with some exception guards and logging. I would prefer we didn't depend on functionality inside scripts, they are not meant to have stable APIs. This is the case where code duplication is preferable to extra dependencies.

Reply:
Updated the code to remove index_datasets